docs/src/HQMTool.md

```diff
@@ -0,0 +1,512 @@
+
+- Parametric equations
+- Splines
+- Lines defined by their end points
+- Circular arcs
+
+In the example, the outer boundary is a closed circular arc with center at [0.0, 0.0, 0.0] with radius 4, starting at zero and ending at 360 degrees. It is added to the project with `addCurveToOuterBoundary!` through the generic name `add!`. You can add any number of curves, but they must be added in order, counter-clockwise.
+
+Similarly, you create curves and add them to as many inner boundaries that you want to have. In the example, there is one inner boundary, "IceCreamCone" made up of two lines and a half circular arc. Again, add them in order, counter-clockwise.
+
+For convenience, `newProject` will generate default run parameters, like the plot file format and the smoother. The parameters can be edited with setter commands. For example, the script sets the polynomial order (default = 5) and the plot file format (default = "skeleton").
+
+One run parameter that must be set manually is the background grid. Since there is an outer boundary, that determines the extent of the domain to be meshed, so only the mesh size needs to be specified using
+
+		addBackgroundGrid!(proj::Project, bgSize::Array{Float64})
+
+The example sets the background mesh size to be 0.1 in the x and y directions. The z component is ignored.
+
+The script finishes by generating the quad mesh and plotting the results, as shown below
+
+![iceCreamCone](https://user-images.githubusercontent.com/3637659/132798939-218a3379-7d50-4f3e-9bec-e75e6cd79031.png)
+
+It also returns the project so that it can be edited further, if desired.
+
+To save a control file for HOHQMesh, simply invoke
+
+		saveProject(proj::Project,outFile::String)
+
+where outFile is the name of the control file (traditionally with a .control extension). `saveProject` is automatically called when a mesh is generated.
+
+The third example `ice_cream_cone_demo` is identical to that which was explained above except that the function calls
+use the generic version of, e.g., `new` or `add!`.
+
+Methods are available to edit a model. For example to move the center of the outer boundary.
+
+## Basic Moves
+
+To create generate a mesh you
+
+- [Create a project](#newProject)
+
+		p = newProject(<projectName>,<folder>)
+
+- [Create inner and outer boundary curves](#DefiningCurves)
+
+		c = new(<name>, startLocation [x,y,z],endLocation [x,y,z])     (Straight Line)
+		c = new(<name>,center [x,y,z],radius,startAngle,endAngle,units = "degrees" or "radians") (Circular Arc)
+		c = new(<name>, xEqn, yEqn, zEqn ) (Parametric equation)
+		c = new(<name>, dataFile) (Spline)
+		c = new(<name>, nKnots, knotsMatrix) (also Spline)
+
+- [Add curves](#AddingCurves) to build the model to see what you have added,
+
+		add!(p, <curveName>) (Add outer boundary curve)
+		add!(p, <curveName>, <InnerBoundaryName>) (add curve to an inner boundary)
+
+- To [visualize](#Plotting) the project's model,
+
+		plotProject!(p,MODEL)
```

Comment:
Does this really modify the project (significantly)? If not, the convention is that `!` shouldn't be there.

Reply:
It modifies the plot object that is carried in the variable `p.plt`.